src/extension-runners/chromium.js

```diff
@@ -0,0 +1,286 @@
+      })
+    );
+
+    const wssInfo = this.wss.address();
+
+    const bgPage = `(function bgPage(webExtParams) {
+      async function getAllDevExtensions() {
+        const [managerExtension, allExtensions] = await Promise.all([
+          new Promise(r => chrome.management.getSelf(r)),
+          new Promise(r => chrome.management.getAll(r)),
+        ]);
+
+        return allExtensions.filter((extension) => {
+          return extension.installType === "development" &&
+            extension.id !== managerExtension.id;
+        });
+      }
+
+      const setEnabled = (extensionId, value) =>
+        new Promise(r => chrome.management.setEnabled(extensionId, value, r));
+
+      async function reloadExtension(extensionId) {
+        await setEnabled(extensionId, false);
+        await setEnabled(extensionId, true);
+      }
+
+      const ws = new window.WebSocket(
+        "ws://${wssInfo.address}:${wssInfo.port}");
+
+      ws.onmessage = async (evt) => {
+        const msg = JSON.parse(evt.data);
+        if (msg.type === 'webExtReloadAllExtensions') {
+          const devExtensions = await getAllDevExtensions();
+          for (var ext of devExtensions) {
+            reloadExtension(ext.id);
+          }
+        }
+      };
+    })()`;
+
+    await fs.writeFile(path.join(extPath, 'bg.js'), bgPage);
+    return extPath;
+  }
+
+  /**
+   * Reloads all the extensions, collect any reload error and resolves to
+   * an array composed by a single ExtensionRunnerReloadResult object.
+   */
+  async reloadAllExtensions(): Promise<Array<ExtensionRunnerReloadResult>> {
+    const runnerName = this.getName();
+
+    // TODO(rpl): file github issue to improve the chromium extension runner
+    // to actually wait for the wssBroadcast to be processed by the connected
+    // client.
+    this.wssBroadcast({
+      type: 'webExtReloadAllExtensions',
+    });
+
+    process.stdout.write(
+      `\rLast extension reload: ${(new Date()).toTimeString()}`);
+    log.debug('\n');
+
+    return [{runnerName}];
+  }
+
+  /**
+   * Reloads a single extension, collect any reload error and resolves to
+   * an array composed by a single ExtensionRunnerReloadResult object.
+   */
+  async reloadExtensionBySourceDir(
+    extensionSourceDir: string // eslint-disable-line no-unused-vars
+  ): Promise<Array<ExtensionRunnerReloadResult>> {
+    // TODO(rpl): file github issue to improve the manager extension
+    // to make it able to detect the extension ids assigned to the
+    // target extensions and map it to the extensions source dir.
+    return this.reloadAllExtensions();
+  }
+
+  /**
+   * Register a callback to be called when the runner has been exited
+   * (e.g. the Chromium instance exits or the user has requested web-ext
+   * to exit).
+   */
+  registerCleanup(fn: Function): void {
+    this.cleanupCallbacks.add(fn);
+  }
+
+  /**
+   * Exits the runner, by closing the managed Chromium instance.
+   */
+  async exit(): Promise<void> {
```

Comment:
Note: The whole implementation is not prepared to handle `.exit()` calls while `run` is still pending. If `exit()` is called while the helper (extension) directories are being set up, or before Chrome has finished launching, then the cleanup won't finish properly.

Is this intentional?

Reply:
I added some additional changes to ensure we wait for the async setup to complete (and additional test to cover that part in the unit tests)